### PR TITLE
[home-assistant] add esphome dependency

### PR DESCRIPTION
**Pull request by hall**
#### Special notes for your reviewer:

These changes deploy the `esphome` chart alongside the `home-assistant` chart and [share home-assistant's `/config/secrets.yaml` with the esphome container](https://esphome.io/guides/faq.html#how-do-i-use-my-home-assistant-secrets-yaml).

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[radarr]`)

### CODE REVIEW

File: charts/home-assistant/values.yaml [hunk 1]
```diff
@@ -322,6 +322,18 @@ appdaemon:
     loadBalancerSourceRanges: []
     # nodePort: 30000
 
+esphome:
+  enabled: false
+  # Mount home-assistant's secrets.yaml into the esphome container.
+  extraVolumes:
+    - name: ha-config
+      persistentVolumeClaim:
+        claimName: home-assistant
```
Comment (hall) [inline charts/home-assistant/values.yaml:331]: This is a little flaky as it depends on the release name but I'm not quite sure how to get around that.

**Comment by billimek** [comment]
I think I know why this is failing the testing automation - this is the first chart with a dependency.  Give me a bit to get this sorted.

**Comment by billimek** [comment]
@hall I probably won't get a chance to look at this until Monday.  But I believe we need a `requirements.yaml` file for this situation.  Can you please take a look at the [teslamate/requirements.yaml](https://github.com/billimek/billimek-charts/blob/master/charts/teslamate/requirements.yaml) and associated files for an example?

**Comment by hall** [comment]
@billimek, no worries -- there's no rush on this! The [`helm/chart-testing-action@v1.0.0-rc.2`](https://github.com/helm/chart-testing-action/tree/v1.0.0-rc.2#chart-testing-action) action claims to only work with Helm 3 (since the `v1.0.0-rc.2` tag points to `master`) and [`requirements.yaml` was deprecated](https://helm.sh/docs/faq/#consolidation-of-requirementsyaml-into-chartyaml). As Helm 3 can still use the v1 chart API, I can update this PR to use `requirements.yaml` but I'm doubting whether that's truly the issue here.

I'm also suspicious of this error:

```
Error: no repository definition for https://billimek.com/billimek-charts. Please add the missing repos via 'helm repo add'
```

That sounds like an intermittent issue. At any rate, review/retry at your leisure. If it still isn't working, I'll just revert back to the v1 chart API.

File: charts/home-assistant/values.yaml (hunk 1 above)
Comment (billimek) [reply charts/home-assistant/values.yaml:331]: We probably need to find a clever way to handle this, as hard-coding to `home-assistant` is less-than-ideal.  I'm poking around about how to communicate a templated name (or equivalent) here instead.

Comment (carpenike) [reply charts/home-assistant/values.yaml:331]: Perhaps using something like? Does the claim always match the release name?

https://github.com/billimek/billimek-charts/blob/master/charts/powerdns/templates/secret.yaml#L10

Comment (billimek) [reply charts/home-assistant/values.yaml:331]: I don't think templates can work inside the `values.yaml` file.  I came across https://github.com/helm/helm/issues/2492 describing pretty much this problem.

It seems that until there is a way to handle templates in the values, we are probably going to just need to hard-code, like we do here.

What do you all think?

**Comment by carpenike** [comment]
Let's merge it so i can PR the DB part. :)

**Pull request merged**